<a href="https://colab.research.google.com/github/ira89l/Home-tasks-/blob/main/hw_1_1_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

def viraz (a,b,c,d):
  result = (a + b) * c / d
  print (result)

viraz (a,b,c,d)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [2]:
rem = 11283 % 89
print (rem)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [3]:
int_div = 11283 // 89
print (int_div)

126


4. Запишіть у змінну `name` ваше імʼя.

In [4]:
name = 'Iryna'
print (name)

Iryna


5. Запишіть у змінну `surname` ваше прізвище.

In [6]:
surname = 'Loshakova'
print (surname)

Loshakova


6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [7]:
name_surname = name + " " + surname
print (name_surname)

Iryna Loshakova


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [8]:
print (name_surname.upper())

IRYNA LOSHAKOVA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [9]:
print (name_surname.lower())

iryna loshakova


8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [10]:
print (name_surname[2])

y


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [12]:
print(surname[0])

def end (surname):
  len_end = len(surname) - 1
  print (surname[len_end])

end (surname)

L
a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [13]:
print (len(surname))

9


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [15]:
capital = 'Antananarivo'

def count_a (capital):
  count = 0
  for i in capital:
    if i == "a" or i == "A":
      count = count + 1
      print (count)

count_a (capital)

1
2
3
4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [19]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print (my_string.startswith('The'))
print (my_string.endswith('dog!'))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [22]:
split = my_string.split()
united = "_".join(split)
print (united)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`